Graylog2/graylog2-server
# Add JUnit 5 with vintage engine alongside JUnit 4

Adding JUnit 5 support to enable use of its features and begin the upgrade process away from JUnit 4. A full upgrade to JUnit 5 doesn't look feasible in a reasonable amount of time (see below).

## Description
- Upgraded JUnit 4 to 4.13
- Added JUnit 5 dependencies to allow writing and running JUnit 5 tests
- Added JUnit 5 vintage engine to allow running a mix of JUnit 4 and 5 tests

## Motivation and Context
JUnit 5 has been out for a while now. While a big upgrade from 4 to 5 would mean a lot of work in the short term, we can begin a smooth migration by adding JUnit 5 alongside 4. New tests can then be written with JUnit 5 while old ones can be migrated at a later point. We might not even be able to get rid of JUnit 4 in the foreseeable future, because some of our test libraries depend on it (see below).

Here's a (probably incomplete) list of to-dos, if we want to upgrade to 5 completely:
 - adapt annotations (e.g., `@Before` -> `BeforeEach`, `@Ignore` ->  `@Disabled`)
- remove public modifiers from test classes and methods (optional, makes the tests more concise)
- replace JUnit 4 parameterized tests with `@ParameterizedTest` (only 8  test classes to change) 
- replace custom `ConditionalRunner` (only used in 2 classes, shouldn't  be hard to replace, see e.g.,
https://www.codeaffine.com/2016/04/06/replace-rules-in-junit5/)
- in general `Rules` have been replaced with `Extensions` so every place we use `@Rule` needs to be changed, likewise: `@RunWith` -> `@ExtendWith`

But a few of our test dependencies depend on junit 4 and won't work with 5 alone:
- jukito
- fongo
- nosqlunit
- rest-assured: not sure about the library itself, but we are integrating it with self-written `@Rules` which have to be converted to extensions
- system-rules
- testcontainers 
    - integrated with self-written rules which should be straightforward to convert to extensions
    - the lib itself depends on junit 4
- apacheds

## How Has This Been Tested?
- Verify execution of old tests with IntelliJ and Maven
- Add a temporary JUnit 5 test class and verify execution from Intellij and Maven
- Added a failing JUnit 5 test temporarily to make sure that everything works on Jenkins and Travis. See failed builds here:
  - Jenkins: https://jenkins.torch.sh/job/graylog-project-pr-snapshot/8223//console
  - Travis: https://travis-ci.org/Graylog2/graylog2-server/builds/643340766?utm_source=github_status&utm_medium=notification

## Screenshots (if appropriate):

## Types of changes
